process/logs/logs_processor.go

```diff
@@ -25,11 +25,11 @@ func NewLogsProcessor(pubKeyConverter core.PubkeyConverter) (*logsProcessor, err
 }
 
 // ProcessLogs converts logs data to a specific structure defined by avro schema
-func (lp *logsProcessor) ProcessLogs(logs map[string]data.LogHandler) []*schema.Log {
+func (lp *logsProcessor) ProcessLogs(logs []*data.LogData) []*schema.Log {
 	allLogs := make([]*schema.Log, 0, len(logs))
 
-	for currHash, currLog := range logs {
-		processedLog := lp.processLog(currHash, currLog)
+	for _, currLog := range logs {
+		processedLog := lp.processLog(*currLog)
```

Comment:
This can panic, please pass this as reference and add a new check on L42

Reply:
fixed